### ISSUE TITLE
Switching environment in the CLI should also change the value of environment variables in the current terminal

### ISSUE THREAD

**Issue by smarie** [smarie/env-switcher-gui#11]
Hints:
- https://stackoverflow.com/questions/171588/is-there-a-command-to-refresh-environment-variables-from-the-command-prompt-in-w
- https://github.com/chocolatey/chocolatey/blob/master/src/redirects/RefreshEnv.cmd

**Comment by jayvdb** [comment]
RefreshEnv.cmd works quite well, but unlikely to help you as you want to cause the parent process to be reset.

https://github.com/tischda/refresh does that, using the broadcast mentioned in the SO you linked to above.
https://github.com/SecsAndCyber/PathFix/blob/master/src/pathfix.py#L11 does it in Python, and there are a few other examples in Python that can be found on GitHub , e.g. https://github.com/WiggleWizard/unrealtoolkit/blob/90dbb11e78314e854ffda55ad18ecfa8bb372436/unrealtoolkit/env.py#L5, but I am not sure exactly what works for those flags - I see a lot of variation.

See also https://github.com/saltstack/salt/issues/23504#issuecomment-106927805 and rest of the thread.

**Comment by jayvdb** [comment]
I've played around with this quite a bit over the last day for my own script, and I am pretty confident that RefreshEnv.cmd is the only way to fix this.

https://github.com/Maximus5/ConEmu/issues/468 also covers some of the reasons why a terminal shouldnt automatically refresh its env vars, and it has a patch very recently uploaded ... ;-)

**Comment by smarie** [comment]
Thanks @jayvdb ! 

I am not sure of the final conclusion though: would there be a chance for a python process (the commandline setuptools entrypoint included in `envswitch`) to perform what RefreshEnv.cmd does ? If so even if it would only work on windows I would gladly accept a PR or an example code snippet.

**Comment by jayvdb** [comment]
`RefreshEnv.cmd` generates a temporary .cmd file and `call`'s it.  We can use python to create the generated .cmd file, but there is no way that python can `call` it -- that has to be done in the running `cmd.exe` process.

The best you could do is tell users 

> Please run `call %TEMP%\foo.cmd` to refresh your current console

But then why not include `RefreshEnv.cmd` and ask them to run it.

**Comment by smarie** [comment]
You are right: the best is probably to include it as a resource and ask them to run it. Thanks !

**Comment by jayvdb** [comment]
> would there be a chance for a python process (the commandline setuptools entrypoint included in envswitch) to perform what RefreshEnv.cmd does ?

Ahhh, yes ... if you can get control of the contents of `envswitch.cmd` , it could invoke `refreshenv`.  I havent tried to override the setuptools .cmd helper.  My guess is that it wont be easy to do during `setup.py`, and would be very tricky to do in a wheel (but see https://github.com/amzn/ion-python/pull/76#issuecomment-471315694 and my comment below it; and also https://github.com/ninjaaron/fast-entry_points), but you could do it at startup which would then be effective for all subsequent usage.

Then you have the problem of being cross platform ... :)  See https://github.com/chocolatey/choco/issues/1851

**Comment by smarie** [comment]
Modifying the setuptools-generated entrypoint is a good idea ! We'll see how this goes when I have a window of time to have a look (or if someone proposes a PR).
